LokiLabs/microsoft-graph-explorer-v4
# Interns/lhao03/38011 create the UI components for permissions

## Overview

* we need a button to switch between user mode and Teams app mode. Right now this button:
* changes the header from "Graph Explorer (as user)" to  "Graph Explorer (as Teams app)".
* changes the blurb in the "Modify Permissions (Preview)" permissions tab from "One of the following delegated permissions is required to run the query. Select a permission and click Consent." from "One of the following resource specific consent permissions is required to run the query. Select a permission and click Consent.".
* changes the permissions from DelegatedWork to Application in the "Modify Permissions (Preview)" tab.
* right now can't accept Application permissions
* accepting delegated permissions still works.

### Demo

![perms](https://user-images.githubusercontent.com/55033656/123186426-f3557880-d44c-11eb-8088-d811a0dfe3be.gif)


### Notes

* right now someone can't use Teams app mode when not signed in. Do we want to keep it that way?

## Testing Instructions

* Start after checking out this branch
* If you're using WSL, run `nvm use 14.0`.
* then run `npm install && npm start`
* everything should work as in the gif.
